### PR TITLE
Refs #25703 - returned some misdeleted host tests

### PR DESCRIPTION
**Pull request by lzap**
Returns the mis-deleted host tests.

Can you @tbrisker please fix possible issues/comments you may have and merge my work even if I won't be responding through the holidays? I am returning back the whole deleted block and changing the five tests to pass. The changes were pretty straighforward except "compute attributes are populated by hardware*" two tests which I struggled a bit and I ran out of time. One is passing although I am not confident and the other does not work. Not sure about those associations.

This is just matter of time, oh no my wife is calling me! I really need to g

END OF TRANSMISSION 2018

### CODE REVIEW

File: test/models/host_test.rb [hunk 1]
```diff
@@ -920,6 +920,1520 @@ def teardown
...
+                             :on_compute_resource,
+                             :with_compute_profile,
+                             :with_ipv6_dns_orchestration,
+                             :ip6 => nil,
+                             :subnet6 => subnet6)
+    host.compute_resource.stubs(:provided_attributes).returns({:mac => :mac})
+    assert_nil host.ip
+    assert_nil host.ip6
+    refute host.require_ip4_validation?, 'should not require ipv4 validation'
+    refute host.require_ip6_validation?, 'should not require ipv6 validation'
```
Comment (tbrisker) [inline test/models/host_test.rb:2421]: This is the test I changed, all other tests were passing already.

**Comment by tbrisker** [comment]
Since comparing on GH between commits while ignoring whitespace is a bit difficult, here is the relevant part of the diff from the original tests:
```diff
diff --git a/test/models/host_test.rb b/test/models/host_test.rb
index 66968d44e..d0e214cfc 100644
@@ -888,17 +920,6 @@ class HostTest < ActiveSupport::TestCase
     end
   end
 
-  context "location or organizations are not enabled" do
-    before do
-      @original_loc, SETTINGS[:locations_enabled] = SETTINGS[:locations_enabled], false
-      @original_org, SETTINGS[:organizations_enabled] = SETTINGS[:organizations_enabled], false
-    end
-
-    after do
-      SETTINGS[:locations_enabled] = @original_loc
-      SETTINGS[:organizations_enabled] = @original_org
-    end
-
   test "should save if root password is undefined when the host is managed and in build mode" do
     Setting[:root_pass] = ''
     host = Host.new :name => "myfullhost", :managed => true, :build => false
@@ -1359,7 +1380,7 @@ class HostTest < ActiveSupport::TestCase
   end
 
   test "should validate pxe loader when provided" do
-      host = Host.create :name => "myhostpxe", :mac => "aabbecddeeff", :ip => "2.3.4.3", :hostgroup => hostgroups(:common), :managed => true, :pxe_loader => "PXELinux BIOS"
+    host = Host.create :name => "myhostpxe", :mac => "aabbecddeeff", :ip => "2.3.4.3", :hostgroup => hostgroups(:common), :managed => true, :pxe_loader => "PXELinux BIOS", :location => taxonomies(:location1), :organization => taxonomies(:organization1)
     assert_equal "x86_64", host.architecture.name
     assert_equal "PXELinux BIOS", host.pxe_loader
     assert_empty host.errors.messages
@@ -1368,8 +1389,8 @@ class HostTest < ActiveSupport::TestCase
 
   test "should save uuid on managed hosts" do
     Setting[:use_uuid_for_certificates] = true
-      host = Host.create :name => "myhost1", :mac => "aabbecddeeff", :ip => "2.3.4.3", :hostgroup => hostgroups(:common), :managed => true
-      assert host.valid?
+    host = Host.create :name => "myhost1", :mac => "aabbecddeeff", :ip => "2.3.4.3", :hostgroup => hostgroups(:common), :managed => true, :location => taxonomies(:location1), :organization => taxonomies(:organization1)
+    assert host.valid?, host.errors.full_messages.to_sentence
     assert !host.new_record?
     assert_not_nil host.certname
     assert_not_equal host.name, host.certname
@@ -2384,30 +2405,34 @@ class HostTest < ActiveSupport::TestCase
   end
 
   test "compute attributes are populated by hardware profile from hostgroup" do
-      # hostgroups(:common) fixture has compute_profiles(:one)
-      host = FactoryBot.build(:host, :managed,
-        :hostgroup => hostgroups(:common),
-        :compute_resource => compute_resources(:ec2),
-        :organization => nil,
-        :location => nil)
+    cr_hostgroup = FactoryBot.build(:compute_resource, :libvirt)
+    cr_host = FactoryBot.build(:compute_resource, :libvirt)
+    hg = FactoryBot.build(:hostgroup, :compute_resource => cr_hostgroup)
+    host = FactoryBot.build(:host, :managed, :provision_method => 'image',
+      :hostgroup => hg,
+      :compute_resource => cr_host)
+    host.stubs(:provision_method_in_capabilities).returns(true)
     host.stubs(:vm_exists?).returns(false)
-      host.expects(:queue_compute_create)
+    host.stubs(:queue_compute_create)
     assert host.valid?, host.errors.full_messages.to_sentence
-      assert_equal compute_attributes(:one).vm_attrs, host.compute_attributes
+    assert_equal cr_hostgroup.attrs, host.compute_attributes
   end
 
   test "compute attributes are populated by hardware profile passed to host" do
+    resource = FactoryBot.create(:libvirt_cr)
+    profile = FactoryBot.create(:compute_profile)
+    attribute = FactoryBot.create(:compute_attribute, :compute_resource => resource, :compute_profile => profile)
+
     host = FactoryBot.build(:host, :managed,
-        :compute_resource => compute_resources(:ec2),
-        :compute_profile => compute_profiles(:two),
-        :organization => nil,
-        :location => nil)
+      :provision_method => 'image',
+      :compute_resource => resource,
+      :compute_profile => profile)
+    host.stubs(:provision_method_in_capabilities).returns(true)
     host.stubs(:vm_exists?).returns(false)
     host.expects(:queue_compute_create)
     assert host.valid?, host.errors.full_messages.to_sentence
-      assert_equal compute_attributes(:three).vm_attrs, host.compute_attributes
+    assert_equal attribute.vm_attrs, host.compute_attributes
   end
-  end # end of context "location or organizations are not enabled"
 
   test "should not save if owner taxonomies do not match host configuration" do
     org1 = FactoryBot.create(:organization)
@@ -2455,7 +2476,7 @@ class HostTest < ActiveSupport::TestCase
     host.provision_method = 'image'
     host.stubs(:capabilities).returns([:build])
     host.valid?
-    assert host.errors[:provision_method].include?('is an unsupported provisioning method')
+    assert_match(/unsupported provisioning method/, host.errors[:provision_method].to_sentence)
   end
 
   test "#provision_method can be updated to implicit value" do

**Review by tbrisker — APPROVED** [review]
Tests are all green, going to squash and merge this after consulting with @orrabin. Thanks @lzap, sorry for ruining your net-LOCs and happy holidays!

**Comment by lzap** [comment]
Thanks @tbrisker for actually doing the work, that was a stressful day. Christmas is over! Yay.

**Pull request merged**